Restructure Settings, enable dark theme - Fix #840

hkedia321 opened the pull request:
Fixes issue #840 

Changes:
- Fix bug in saving dark theme while signed in - fixes #841 
- Show User's email address
- Remove unnecessary popups in Settings page
- Disable refreshing after changing settings
- Save changes only for the changed settings of the current tab
- Enable dark theme in Settings page
- Integrate Mobile page

Demo Link: [clammy-park.surge.sh](http://clammy-park.surge.sh)

Screenshots for the change: 
![image](https://user-images.githubusercontent.com/17807257/31841739-7d687700-b608-11e7-9f92-793b2324c67b.png)
![image](https://user-images.githubusercontent.com/17807257/31841777-a0c8862c-b608-11e7-83d6-26893e3d6a48.png)

hkedia321 commented:
@isuruAb Now changing the settings is working. Please check. Let me know if any changes are required.

isuruAb commented on src/components/ChatApp/Settings/Settings.react.js:
```diff
@@ -227,7 +249,7 @@ class Settings extends Component {
 	}
 	this.props.history.push(`/settings?tab=${this.state.selectedSetting}`);
-	window.location.reload();
+	// window.location.reload();
```
please remove unnecessary  comments

isuruAb commented on src/actions/API.actions.js:
```diff
@@ -365,6 +367,10 @@ export function getSettings(){
       timeout: 3000,
       async: false,
       success: function (response) {
+          console.log(response)
```
please remove unwanted logs 👍

hkedia321 commented:
@isuruAb @madhavrathi I have removed the unwanted logs and comments. :+1:  Thanks!

hkedia321 commented:
@mariobehling I have changed the required text and updated the surge (link)[http://breakable-kittens.surge.sh]. Please let me know if any more changes are required. :+1:

isuruAb commented:
@hkedia321  please resolve conflicts

hkedia321 commented:
@isuruAb @mariobehling I have resolved the merge conflicts, and have updated the screenshots and surge [link](http://untidy-roof.surge.sh/). Please have a look.

mariobehling commented:
Good job! Unfortunately there was another PR in the queue that resulted in another conflict. Please help. Thanks.

isuruAb commented:
@hkedia321  please resolve conflicts 👍

hkedia321 commented:
@mariobehling @isuruAb Resolved the merge conflicts! Here is the updated surge: [clammy-park.surge.sh](http://clammy-park.surge.sh). Please review.

mariobehling commented:
Thanks! Good job.
